ui/src/components/Investor/Investor.tsx

```diff
@@ -45,6 +47,38 @@ const Investor: React.FC<Props> = ({ onLogout }) => {
     const allDeposits = useStreamQuery(AssetDeposit).contracts.map(makeContractInfo);
     const allCustodianRelationships = useStreamQuery(CustodianRelationship).contracts.map(makeContractInfo);
 
+    const { brokerMap, custodianMap, exchangeMap } = useRegistryLookup();
+
+    const brokerProviders = useStreamQuery(BrokerCustomer).contracts
+        .map(broker => {
+            const party = broker.payload.broker;
+            const name = brokerMap.get(damlTupleToString(broker.key))?.name;
+            return {
+                party,
+                label: `${name ? `${name} (${party})` : party} | Broker`,
+            }
+        })
+
+    const allProviders = [
+        ...allExchanges.map(exchange => {
```

Comment:
would it make more sense here to use the `ExchangeParticipant` contracts (just like you do with the `BrokerCustomer`) to find the available exchanges that the investor has a relationship with?